library/system/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,301 @@
+    # @raise ArgumentError when mode is not valid
+    def start_mode=(mode)
+      if !start_modes.include?(mode)
+        raise ArgumentError, "Invalid start mode: '#{mode}' for service '#{service.name}'"
+      end
+
+      if mode == current_start_mode
+        unregister_change(:start_mode)
+      else
+        register_change(:start_mode, mode)
+      end
+    end
+
+    # Determine whether the service will be active after calling #save
+    #
+    # @return [Boolean] true if the service must be active; false otherwise
+    def active?
+      return new_value_for(:active) if changed_value?(:active)
+      service.active?
+    end
+
+    # Sets the service to be started after calling #save
+    #
+    # @see #active=
+    def start
+      self.active = true
+    end
+
+    # Sets the service to be stopped after calling #save
+    #
+    # @see #active=
+    def stop
+      self.active = false
+    end
+
+    # Toggles the service status
+    #
+    # @see #active=
+    def toggle
+      self.active = !active?
+    end
+
+    # Saves changes to the underlying system
+    #
+    # @param set_status [Boolean] Do not change service status. Useful when running on 1st stage.
+    def save(ignore_status: false)
+      save_start_mode
+      set_current_status unless ignore_status
+      reload
+    end
+
+    # Reverts stored changes
+    def reload
+      clear_changes
+      @current_start_mode = nil
+    end
+
+    # Determines whether the system has been changed or not
+    #
+    # @return [Boolean] true if the system has been changed
+    def changed?
+      !changes.empty?
+    end
+
+    # Returns the list of supported start modes
+    #
+    # * :on_boot:   The service will be started when the system boots.
+    # * :manual: The service is disabled and it will be started manually.
+    # * :on_demand: The service will be started on demand (using a Systemd socket).
+    #
+    # @return [Array<Symbol>] List of supported modes.
+    def start_modes
+      return @start_modes if @start_modes
+      @start_modes = [:on_boot, :manual]
+      @start_modes << :on_demand if socket
+      @start_modes
+    end
+
+    # Terms to search for this service
+    #
+    # In case the service has an associated socket, the socket name
+    # is included as search term.
+    #
+    # @return [Array<String>] e.g., #=> ["tftp.service", "tftp.socket"]
+    def search_terms
+      terms = [service.id]
+      terms << socket.id if socket
+      terms
+    end
+
+    # Determines whether a value has been changed
+    #
+    # @return [Boolean] true if the value has been changed; false otherwise
+    def changed_value?(key)
+      changes.key?(key)
+    end
+
+  private
+
+    # @return [Hash<String,Object>]
+    attr_reader :changes
+
+    # Sets whether the service should be active or not
+    #
+    # The given value will be applied after calling #save.
+    #
+    # @param value [Boolean] true to set this service as active
+    def active=(value)
```

Comment:
to be honest, I really dislike this API, as it ignore that we maybe want also restart or reload. I think we also discuss that this ability will be useful also for services manager to allow restart or reload.
What I do not like is that you set target state, not action that should be done on top of it.